jenkinsci/blueocean-plugin
# [JENKINS-47929] Allow extensible run details tabs

# Description

See [JENKINS-47929](https://issues.jenkins-ci.org/browse/JENKINS-47929).
Reviving [PR#2050](https://github.com/jenkinsci/blueocean-plugin/pull/2050).

**Unit/Acceptance Tests**
AFAICT the existing tests provide enough coverage for this change as the default actions (changes, tests, artifacts) have been modified to use the new extension point.

I have created the [Blue Ocean Build Failure Analyzer Plugin](https://github.com/stuartrowe/blueocean-bfa-plugin) as a proof of concept. This plugin demonstrates how to extend the `jenkins.run.actions` extension point. The screenshot below shows a new `Failure Causes` tab and the `Found Failure Causes` identified by the [Build Failure Analyzer Plugin](https://github.com/jenkinsci/build-failure-analyzer-plugin) scan.

![image](https://user-images.githubusercontent.com/551572/161340307-6a38c2f5-d6d4-4748-ac90-7da68738d533.png)


**Manual Test Instructions**
- Build and run the changes to verify that the changes, tests and artifacts action tabs still work.
- Optionally clone, build and install the [blueocean-bfa-plugin](https://github.com/stuartrowe/blueocean-bfa-plugin) to verify how external plugins can extend the Run Details view.

# Submitter checklist

# Reviewer checklist
